engine/access/rpc/backend/accounts/accounts.go

```diff
@@ -0,0 +1,238 @@
+package accounts
+
+import (
+	"context"
+
+	"github.com/rs/zerolog"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/onflow/flow-go/engine/access/rpc/backend"
+	"github.com/onflow/flow-go/engine/access/rpc/backend/accounts/handler"
+	"github.com/onflow/flow-go/engine/access/rpc/connection"
+	commonrpc "github.com/onflow/flow-go/engine/common/rpc"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/execution"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/state/protocol"
+	"github.com/onflow/flow-go/storage"
+)
+
+type Accounts struct {
+	log             zerolog.Logger
+	state           protocol.State
+	headers         storage.Headers
+	endpointHandler handler.Handler
+}
+
+func NewAccounts(
+	log zerolog.Logger,
+	state protocol.State,
+	headers storage.Headers,
+	connFactory connection.ConnectionFactory,
+	nodeCommunicator backend.Communicator,
+	scriptExecMode backend.IndexQueryMode,
+	scriptExecutor execution.ScriptExecutor,
+	execNodeIdentitiesProvider *commonrpc.ExecutionNodeIdentitiesProvider,
+) (*Accounts, error) {
+	var h handler.Handler
+
+	// TODO: we can instantiate these strategies outside of backend (in access_node_builder e.g.)
+	switch scriptExecMode {
+	case backend.IndexQueryModeLocalOnly:
+		h = handler.NewLocalHandler(log, state, scriptExecutor)
+
+	case backend.IndexQueryModeExecutionNodesOnly:
+		h = handler.NewExecutionNodeHandler(log, state, connFactory, nodeCommunicator, execNodeIdentitiesProvider)
+
+	case backend.IndexQueryModeFailover:
+		local := handler.NewLocalHandler(log, state, scriptExecutor)
+		execNode := handler.NewExecutionNodeHandler(log, state, connFactory, nodeCommunicator, execNodeIdentitiesProvider)
+		h = handler.NewFailoverHandler(log, state, local, execNode)
+
+	case backend.IndexQueryModeCompare:
```

Comment:
we can drop the compare options in the new version since they're not needed anymore. feel free to do that in a separate PR if that's easier